CoreTableView/test/unit/src/au/gov/asd/tac/constellation/views/tableview2/io/TableViewPreferencesIOUtilitiesNGTest.java

```diff
@@ -0,0 +1,214 @@
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import javafx.collections.ObservableList;
+import javafx.embed.swing.JFXPanel;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import org.openide.util.NbPreferences;
+import static org.testng.AssertJUnit.assertEquals;
+import org.testng.SkipException;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author formalhaunt
+ */
+public class TableViewPreferencesIOUtilitiesNGTest {
+    private static MockedStatic<JsonIO> jsonIOStaticMock;
+    private static MockedStatic<NbPreferences> nbPreferencesStaticMock;
+    private static MockedStatic<ApplicationPreferenceKeys> applicationPrefKeysStaticMock;
+    
+    public TableViewPreferencesIOUtilitiesNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        jsonIOStaticMock = Mockito.mockStatic(JsonIO.class);
+        nbPreferencesStaticMock = Mockito.mockStatic(NbPreferences.class);
+        applicationPrefKeysStaticMock = Mockito.mockStatic(ApplicationPreferenceKeys.class);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        jsonIOStaticMock.close();
+        nbPreferencesStaticMock.close();
+        applicationPrefKeysStaticMock.close();
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        jsonIOStaticMock.reset();
+        nbPreferencesStaticMock.reset();
+        applicationPrefKeysStaticMock.reset();
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+    
+    @Test
+    public void getPreferencesMultiplePrefsPicksLast() throws IOException {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        final JsonNode jsonNode = objectMapper.readTree(
+                new FileInputStream(getClass().getResource("resources/vertex-preferences.json").getPath())
+        );
+        
+        jsonIOStaticMock.when(() -> JsonIO.loadJsonPreferences("TableViewPreferences", "vertex-"))
+                .thenReturn(jsonNode);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> preferences
+                = TableViewPreferencesIOUtilities.getPreferences(GraphElementType.VERTEX, 2);
+        
+        final ThreeTuple<List<String>, Tuple<String, TableColumn.SortType>, Integer> expected = ThreeTuple.create(
+                List.of("ABC", "DEF", "ABC", "DEF"), // <- This seems wrong??
```

Comment:
I'd be curious to take a further look at this

Reply:
This is happening because the code is only ever expecting a single object in the array. Which if thats all there is meant to be, I guess begs the question, should it be an array??

I am just writing the tests based on existing behaviour. I am unsure what this is meant to be or if this is correct. Need so guidance from one of the authors.